src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Application {
+    public static void main(String[] args) {
+        int figureNumber = 11;
+
+        int figuresNumberInSetOne = 0;
+
+        if (figureNumber % 2 == 0) {
+            figuresNumberInSetOne = figureNumber / 2;
+        } else {
+            figuresNumberInSetOne = figureNumber / 2 + 1;
+        }
+
+        Figure [] result = new Figure[figureNumber];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < figuresNumberInSetOne; i++) {
+            result [i] = figureSupplier.getRandomFigure();
+            result [i].draw();
+        }
+
+        for (int i = figuresNumberInSetOne; i < figureNumber; i++) {
+            result [i] = figureSupplier.getDefaultFigure();
+            result [i].draw();

```

Comment:
The 'draw()' method is called directly after creating the figure, without checking if the figure was created successfully or not. This can lead to a NullPointerException.